specification/monitor/resource-manager/Microsoft.Insights/preview/2021-05-01-preview/diagnosticsSettingsCategories_API.json

```diff
@@ -0,0 +1,284 @@
+        "x-ms-examples": {
+          "Gets the diagnostic setting": {
+            "$ref": "./examples/listDiagnosticSettingsCategories.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ProxyOnlyResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "A proxy only azure resource object"
+    },
+    "DiagnosticSettingsCategory": {
+      "properties": {
+        "categoryType": {
+          "enum": [
+            "Metrics",
+            "Logs"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "categoryType"
+          },
+          "x-nullable": false,
+          "description": "The type of the diagnostic settings category."
+        },
+        "categoryGroups": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "the collection of what category groups are supported."
+        }
+      },
+      "description": "The diagnostic settings Category."
+    },
+    "DiagnosticSettingsCategoryResource": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DiagnosticSettingsCategory",
+          "description": "The properties of a Diagnostic Settings Category."
+        },
+        "systemData": {
+          "readOnly": true,
+          "description": "The system metadata related to this resource.",
+          "$ref": "#/definitions/SystemData"
+        }
+      },
+      "description": "The diagnostic settings category resource."
+    },
+    "DiagnosticSettingsCategoryResourceCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DiagnosticSettingsCategoryResource"
+          },
+          "description": "The collection of diagnostic settings category resources."
+        }
+      },
+      "description": "Represents a collection of diagnostic setting category resources."
+    },
+    "ErrorResponse": {
+      "description": "Describes the format of Error response.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "Error code",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message indicating why the operation failed.",
+          "type": "string"
+        }
+      }
+    },
+    "SystemData": {
```

Comment:
>  "SystemData" [](http://example.com/codeflow?start=2&length=14)

Consider referencing the common type instead of copying.
[https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json)